### PR TITLE
Update ClientSession reference

### LINKED ISSUE #7515: ClientSession TypeError: max_line_size and max_field_size args
### Describe the bug

max_line_size and max_field_size args for ClientSession are returning TypeError.

### To Reproduce

```
import aiohttp
import asyncio

async def fetch(client):
    async with client.get('http://python.org') as resp:
        assert resp.status == 200
        return await resp.text()

async def main():
    async with aiohttp.ClientSession(max_line_size=8190*2) as client:
        html = await fetch(client)
        print(html)

asyncio.run(main())
```

Used code snippet from [documentation](https://docs.aiohttp.org/en/stable/client_reference.html#aiohttp.ClientSession) and added: `max_line_size=8190*2` for this example.

Which returns:
`TypeError: ClientSession.__init__() got an unexpected keyword argument 'max_line_size'`

This also applied to `max_field_size`.

Despite the options being in the codebase: https://github.com/aio-libs/aiohttp/blob/bc37e44f233ab8f5ed543ba842b1a4866cd77252/aiohttp/client.py#L222

But it is also missing from the official documentation:
https://docs.aiohttp.org/en/stable/_modules/aiohttp/client.html#ClientSession
https://docs.aiohttp.org/en/stable/client_reference.html#aiohttp.ClientSession

Which suggests this also needs updating. I may be mistaken, but I looks like the [tests](https://github.com/aio-libs/aiohttp/commit/ecd9c72697ca1529e7cae4693775612987a2a6d2) when these args were added do not cover the client side. 

### Expected behavior

No TypeError, but being able to define a larger accepted max_line_size and max_field_size when reading response headers from other websites. 

### Logs/tracebacks

```python-traceback
Traceback (most recent call last):
  File "/aiohttp_test/test.py", line 14, in <module>
    asyncio.run(main())
  File "/usr/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/aiohttp_test/test.py", line 10, in main
    async with aiohttp.ClientSession(max_line_size=8190*2) as client:
TypeError: ClientSession.__init__() got an unexpected keyword argument 'max_line_size'
```

### Python Version

```console
$ python --version
Python 3.10.12
```

### aiohttp Version

```console
$ python -m pip show aiohttp
Name: aiohttp
Version: 3.8.5
Summary: Async http client/server framework (asyncio)
Home-page: https://github.com/aio-libs/aiohttp
Author: 
Author-email: 
License: Apache 2
Location: /aiohttp_test/env/lib/python3.10/site-packages
Requires: aiosignal, async-timeout, attrs, charset-normalizer, frozenlist, multidict, yarl
Required-by:
```

### multidict Version

```console
$ python -m pip show multidict
Name: multidict
Version: 6.0.4
Summary: multidict implementation
Home-page: https://github.com/aio-libs/multidict
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache 2
Location: /aiohttp_test/env/lib/python3.10/site-packages
Requires: 
Required-by: aiohttp, yarl
```

### yarl Version

```console
$ python -m pip show yarl
Name: yarl
Version: 1.9.2
Summary: Yet another URL library
Home-page: https://github.com/aio-libs/yarl/
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache-2.0
Location: /aiohttp_test/env/lib/python3.10/site-packages
Requires: idna, multidict
Required-by: aiohttp
```

### OS

Linux firefly 5.19.0-50-generic #50-Ubuntu SMP PREEMPT_DYNAMIC Mon Jul 10 18:24:29 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

### Related component

Client

### Additional context

Trying to crawl a page which hits the max_line_size/max_field_size limit in their headers. As I can't up the number now due to this bug, I can't access the page (even though in Chrome I can)

### Code of Conduct

- [X] I agree to follow the aio-libs Code of Conduct

### PR DESCRIPTION
**Pull request by Dreamsorcerer**
Fixes #7515.

### CODE REVIEW

File: docs/client_reference.rst [hunk 1]
```diff
@@ -41,17 +41,21 @@ The client session supports the context manager protocol for self closing.
                          connector=None, cookies=None, \
                          headers=None, skip_auto_headers=None, \
                          auth=None, json_serialize=json.dumps, \
+                         request_class=ClientRequest, \
+                         response_class=ClientResponse, \
+                         ws_response_class=ClientWebSocketResponse, \
                          version=aiohttp.HttpVersion11, \
                          cookie_jar=None, read_timeout=None, \
```
Comment (Dreamsorcerer) [inline docs/client_reference.rst]:
```suggestion
                         cookie_jar=None, \
```

File: docs/client_reference.rst [hunk 2]
```diff
@@ -106,6 +99,16 @@ The client session supports the context manager protocol for self closing.
...
 
+   :param collections.abc.Callable json_serialize: Json *serializer* callable.
+
+      By default :func:`json.dumps` function.
+
+   :param ClientRequest request_class: Custom class to use for client requests.
+
+   :param ClientRequest response_class: Custom class to use for client responses.
+
+   :param ClientRequest ws_response_class: Custom class to use for websocket responses.
```
Comment (Dreamsorcerer) [inline docs/client_reference.rst]:
```suggestion
   :param ClientResponse response_class: Custom class to use for client responses.

   :param ClientWebSocketResponse ws_response_class: Custom class to use for websocket responses.
```

File: docs/client_reference.rst [hunk 3]
```diff
@@ -106,6 +99,16 @@ The client session supports the context manager protocol for self closing.
                                   otherwise, the default auth will not be
                                   included.
 
+   :param collections.abc.Callable json_serialize: Json *serializer* callable.
+
+      By default :func:`json.dumps` function.
+
+   :param ClientRequest request_class: Custom class to use for client requests.
+
+   :param ClientRequest response_class: Custom class to use for client responses.
```
Comment (bdraco) [inline docs/client_reference.rst]: Should this be a different class?

File: docs/client_reference.rst (hunk 2 above)
Comment (bdraco) [inline docs/client_reference.rst]: Should this be a different class?

File: docs/client_reference.rst (hunk 3 above)
Comment (Dreamsorcerer) [reply docs/client_reference.rst]: Already got it. :P

File: docs/client_reference.rst [hunk 4]
```diff
@@ -106,6 +99,16 @@ The client session supports the context manager protocol for self closing.
                                   otherwise, the default auth will not be
                                   included.
 
+   :param collections.abc.Callable json_serialize: Json *serializer* callable.
+
+      By default :func:`json.dumps` function.
+
+   :param ClientRequest request_class: Custom class to use for client requests.
```
Comment (Dreamsorcerer) [inline docs/client_reference.rst]:
```suggestion
   :param aiohttp.ClientRequest request_class: Custom class to use for client requests.
```

**Pull request merged**